modules/l4proxy/upstream.go

```diff
@@ -30,12 +30,23 @@ import (
 	"github.com/mholt/caddy-l4/layer4"
 )
 
+func ParseAddress(addr string) (*caddy.NetworkAddress, error) {
```

Comment:
Can we add a godoc to this? to explain why it is needed? (I'm not entirely sure myself, just by looking at it, since it uses only exported methods)